ipykernel/ipkernel.py

```diff
@@ -45,7 +45,7 @@
 
 def _create_comm(*args, **kwargs):
     """Create a new Comm."""
-    return BaseComm(*args, **kwargs)
```

Comment:
Ah, I misunderstood what you intended to change.  We wanted to explicitly return `BaseComm` here because it is more efficient.   I think the right answer is for `ipywidgets` to relax its expected trait.

Reply:
yeah, BaseComm is 'the new thing', Comm is for backwards compatibility with ipywidgets.

Reply:
I see, but since `ipywidgets` doesn't accept `BaseComm` should that be the submitted change?  I guess I'm confused because @blink1073's suggestion seemed straightforward, but now I don't understand how to interpret @maartenbreddels's reply.